### ISSUE TITLE
What value for MULTI to use relic multithreaded with Windows

### ISSUE THREAD

**Issue by oliviervibe** [relic-toolkit/relic#229]
What value must be defined for MULTI to use multithreading with Windows?

The 2 values that can be defined - PTHREAD, OPENMP - don't seem to be compatible with Windows!

**Comment by dfaranha** [comment]
Thanks for opening the issue!

When you try one of these, do you see an error message?

**Comment by oliviervibe** [comment]
Thanks for answering so quickly!

Yes, I tried both and none is working. Under Windows there is no pthread.h or omp.h. Windows has its own threading API and model.

For the moment even using it in a multi-threaded application I haven't seen any error yet, but it could happen!

Thanks

Olivier


________________________________

**Comment by dfaranha** [comment]
I will add a CI target for Windows multithreading and see what happens. But I would be very surprised if there's no OpenMP support in Windows.

**Comment by oliviervibe** [comment]
THere is OpenMP on Windows, a colleague confirmed it this weekend.
But the switch /openmp is not set with OPENMP for MULTI.

Thanks

Olivier Rouit




________________________________

**Comment by dfaranha** [comment]
What compiler are you using? The `/openmp` flag is added if MSVC is detected by CMake.

**Comment by dfaranha** [comment]
It now builds correctly, as detected by the CI: https://github.com/relic-toolkit/relic/runs/6049937844?check_suite_focus=true

**Comment by oliviervibe** [comment]
Hi Diego,

I'm using Visual Studio 2019 at the moment for Windows and on other systems VScode with the local toolchains (Linux, MacOS).

We also compile for Android, iOS, and for ARM architecture (with no standard OS).

I also ran in conflicting #define with another library and Windows. Relic defined DOUBLE and it's a very common name in programming! I used RLC_DOUBLE following your naming conventions. You possibly can integrate it as it's very straightforward!

Currently, I keep a local copy of Relic as I want to avoid another issue if the pairing maths of Baretto are changed again.

Thanks a lot,


Olivier Rouit
